ios/samples/hello-ar/hello-ar/FilamentArView/FilamentApp.cpp

```diff
@@ -69,6 +70,90 @@ void FilamentApp::setObjectTransform(const mat4f& transform) {
     meshTransform = transform;
 }
 
+void FilamentApp::updatePlaneGeometry(const FilamentArPlaneGeometry& geometry) {
+    auto& tcm = engine->getTransformManager();
+
+    if (!app.planeGeometry.isNull()) {
+        scene->remove(app.planeGeometry);
+        engine->destroy(app.planeGeometry);
+        tcm.destroy(app.planeGeometry);
+        EntityManager::get().destroy(1, &app.planeGeometry);
+    }
+
+    if (app.planeVertices) {
+        engine->destroy(app.planeVertices);
+    }
+
+    if (app.planeIndices) {
+        engine->destroy(app.planeIndices);
+    }
+
+    if (!app.shadowPlane) {
+        app.shadowPlane = Material::Builder()
+            .package(RESOURCES_SHADOW_PLANE_DATA, RESOURCES_SHADOW_PLANE_SIZE)
+            .build(*engine);
+    }
+
+    float3* normals = new float3[geometry.vertexCount];
+    for (size_t i = 0; i < geometry.vertexCount; i++) {
+        normals[i] = { 0, 1, 0 };
```

Comment:
Since this is a plane you could simply provide SurfaceOrientation with a single normal vector (allocated on the stack), then duplicate the resulting quaternion.  (I think zero stride won't work because it interprets 0 as "tightly packed".)